python/ray/train/_internal/session.py

```diff
@@ -69,7 +71,8 @@ def __init__(
         dataset_shard: Optional[Union[Dataset, DatasetPipeline]] = None,
         # TODO(xwjiang): Legacy Ray Train trainer clean up!
         checkpoint: Optional[Union[Dict, Checkpoint]] = None,
-        encode_data_fn: Callable = None,
+        # Deprecated
+        encode_data_fn: Optional[Callable] = None,
         detailed_autofilled_metrics: bool = False,
     ):
 
```

Comment:
the type hint here https://github.com/ray-project/ray/pull/28794/files#diff-570f30824e8e4ad93529a34c905c7706552621403c1d857224960769d82f9606L83 can now just be `Checkpoint` right?